src/zoo.js

```diff
@@ -11,57 +11,137 @@ eslint no-unused-vars: [
+    newObj[animal.name].sort();
+  }
+  return newObj;
+};
 
-function isManager(id) {
-  // seu código aqui
-}
+const animalMap = function (options) {
+  const obj = { NE: [], NW: [], SE: [], SW: [] };
 
-function addEmployee(id, firstName, lastName, managers, responsibleFor) {
-  // seu código aqui
-}
+  if (!options || !options.includeNames) {
+    data.animals.map(animal => obj[animal.location].push(animal.name));
+  } else {
+    data.animals.map(animal => obj[animal.location].push(createNewObj(animal, options)));
+  }
 
-function animalCount(species) {
-  // seu código aqui
-}
+  return obj;
+};
 
-function entryCalculator(entrants) {
-  // seu código aqui
-}
+const schedule = function (dayName) {
+  const keys = Object.keys(data.hours);
+  const obj = {};
+  keys.forEach(function (cur) {
+    if (data.hours[cur].open === 0) {
+      obj[cur] = 'CLOSED';
+    } else {
+      obj[cur] = `Open from ${data.hours[cur].open}am until ${data.hours[cur].close - 12}pm`;
+    }
+  });
+  const objFiltered = {};
+  objFiltered[dayName] = obj[dayName];
 
-function animalMap(options) {
-  // seu código aqui
-}
+  return (dayName) ? objFiltered : obj;
+};
 
-function schedule(dayName) {
-  // seu código aqui
-}
+const oldestFromFirstSpecies = (id) => {
+  const idAnimal = data.employees.filter(employee => employee.id === id)[0].responsibleFor[0];
+  const Animal = data.animals.filter(animal => animal.id === idAnimal);
+  const olderAnimal = Object.values(Animal[0].residents.sort((a, b) => b.age - a.age)[0]);
+  return olderAnimal;
+};
 
-function oldestFromFirstSpecies(id) {
-  // seu código aqui
-}
+const increasePrices = (percentage) => {
+  const adult = data.prices.Adult;
+  const senior = data.prices.Senior;
+  const child = data.prices.Child;
 
-function increasePrices(percentage) {
-  // seu código aqui
-}
+  data.prices.Adult = Math.round(adult * ((percentage / 100) + 1) * 100) / 100;
+  data.prices.Senior = Math.round(senior * ((percentage / 100) + 1) * 100) / 100;
+  data.prices.Child = Math.round(child * ((percentage / 100) + 1) * 100) / 100;
+};
+
+const createEmployeeObj = () => {
+  const obj = {};
+  data.employees.map((employee) => {
+    const animalNames = [];
+    employee.responsibleFor.map(respId => data.animals.forEach((animal) => {
+      if (respId === animal.id) { animalNames.push(animal.name); }
+    }));
+    obj[`${employee.firstName} ${employee.lastName}`] = animalNames;
+    return animalNames;
+  });
+  return obj;
+};
 
-function employeeCoverage(idOrName) {
-  // seu código aqui
-}
+const employeeCoverage = (ent) => {
+  const obj = createEmployeeObj();
+  if (!ent) {
+    return obj;
+  }
+  const objId = {};
+  data.employees.forEach((employee) => {
+    if (employee.id === ent || employee.firstName === ent || employee.lastName === ent) {
+      const value = obj[`${employee.firstName} ${employee.lastName}`];
+      objId[`${employee.firstName} ${employee.lastName}`] = value;
+    }
+  });
+  return objId;
+};
```

Comment:
Acho que usar object destructuring fica bem massa.

```suggestion
  const objId = {};
  data.employees.forEach(({ id, firstName, lastName }) => {
    if (id === ent || firstName === ent || lastName === ent) {
      const value = obj[`${firstName} ${lastName}`];
      objId[`${firstName} ${lastName}`] = value;
    }
  });
  return objId;
};
```

Reply:
> Acho que usar object destructuring fica bem massa.

Total, fica mais bonito e limpo a leitura!